app/bundles/ProjectBundle/Model/ProjectModel.php

```diff
@@ -4,17 +4,65 @@
 
 namespace Mautic\ProjectBundle\Model;
 
+use Doctrine\ORM\EntityManagerInterface;
+use Mautic\CoreBundle\Helper\CoreParametersHelper;
+use Mautic\CoreBundle\Helper\UserHelper;
+use Mautic\CoreBundle\Model\AjaxLookupModelInterface;
 use Mautic\CoreBundle\Model\FormModel;
+use Mautic\CoreBundle\Security\Permissions\CorePermissions;
+use Mautic\CoreBundle\Translation\Translator;
 use Mautic\ProjectBundle\Entity\Project;
 use Mautic\ProjectBundle\Entity\ProjectRepository;
+use Mautic\ProjectBundle\Service\ProjectEntityLoaderService;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
 
-final class ProjectModel extends FormModel
+final class ProjectModel extends FormModel implements AjaxLookupModelInterface
 {
+    public function __construct(
+        EntityManagerInterface $em,
+        CorePermissions $security,
+        EventDispatcherInterface $dispatcher,
+        UrlGeneratorInterface $router,
+        Translator $translator,
+        UserHelper $userHelper,
+        LoggerInterface $logger,
+        CoreParametersHelper $coreParametersHelper,
+        private ProjectEntityLoaderService $entityLoaderService,
+    ) {
+        parent::__construct($em, $security, $dispatcher, $router, $translator, $userHelper, $logger, $coreParametersHelper);
+    }
+
     public function getRepository(): ProjectRepository
     {
         $repository = $this->em->getRepository(Project::class);
         \assert($repository instanceof ProjectRepository);
 
         return $repository;
     }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @param mixed $type
+     * @param mixed $filter
+     * @param mixed $limit
+     * @param mixed $start
+     *
+     * @return array<int|string, string>
+     */
+    public function getLookupResults($type, $filter = '', $limit = 10, $start = 0): array
```

Comment:
Can we set the right types? I think PHP allows that even if the interface method signature doesn't have any. `mixed` shouldn't be abused.